src/app/plugins/connectfour.plugin.ts

```diff
@@ -204,38 +202,36 @@ class ConnectFourGame {
     return;
   }
 
-  // Turn on the beast.
   private _lionMove() {
+    const bestCol = this._getBestMove();
+    this._dropPiece(bestCol);
+  }
+
+  // Turn on the beast.
+  private _getBestMove(): number {
     const moves: { col: number; val: number }[] = [];
     for (let col = 0; col < this._cols; col++) {
       if (!this._dropPiece(col)) {
```

Comment:
does `this._dropPiece` affect the state of the board?

Reply:
oh, i see the `this._removeTopPiece(col);`
You should prob comment that this change gets reversed at the bottom of the function

Reply:
Yeah I can notate that this is a backtracking algorithm.